File: 226-Invert Binary Tree.py
# 递归
# Runtime: 32 ms, faster than 100.00% of Python3 online submissions for Invert Binary Tree.
# Memory Usage: 12.4 MB, less than 0.96% of Python3 online submissions for Invert Binary Tree.
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def invertTree(self, root: 'TreeNode') -> 'TreeNode':
        if root: root.left, root.right = self.invertTree(root.right), self.invertTree(root.left)
        return root

File: Algorithms/C++/226-Invert_Binary_Tree.cpp
// Runtime: 4 ms, faster than 58.98% of C++ online submissions for Invert Binary Tree.
// Memory Usage: 9.2 MB, less than 85.45% of C++ online submissions for Invert Binary Tree.
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (!root)
            return root;
        swap(root->left, root->right);
        invertTree(root->left);
        invertTree(root->right);
        return root;
    }
};
